src/bokeh/core/serialization.py

```diff
@@ -65,6 +64,8 @@
 from .types import ID
 
 if TYPE_CHECKING:
+    from typing_extensions import NotRequired, TypeAlias
```

Comment:
would it make sense to do this thoughout the codebase?

Reply:
@mattpap If we did that we could potentially drop `typing_extensions` as a *runtime* requirement entirely, no?

Reply:
If the checks pass, I'll let you decide.

While I have your attention, would you have any idea on the issue with https://github.com/kernc/backtesting.py/actions/runs/3301867863/jobs/5447910367#step:9:3? :sweat_smile:

Reply:
@kernc  looks like some migrations to handle, let's take those discussions up on the discourse https://discourse.bokeh.org/